streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -2330,4 +2343,121 @@ public void shouldNotAllowNullNamedOnProcessWithStores() {
+    public void shouldKTableFromKStream() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String())
+            .withName("table-process");
+        final Consumed<String, String> consumed2 = Consumed.with(Serdes.String(), Serdes.String())
+            .withName("stream-table-process");
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input = "input";
+        final String streamToTable = "streamToTableInput";
+        final String output = "output";
+        final String streamToTableOutput = "streamToTableOutput";
+
+        builder.table(input, consumed).toStream().to(output);
+        builder.stream(streamToTable, consumed2).toTable().toStream().to(streamToTableOutput);
+
+        try (final TopologyTestDriver driver = new TopologyTestDriver(builder.build(), props)) {
+            final TestInputTopic<String, String> inputTopic =
+                driver.createInputTopic(input, Serdes.String().serializer(), Serdes.String().serializer());
+            final TestInputTopic<String, String> streamToTableInputTopic1 =
+                driver.createInputTopic(streamToTable, Serdes.String().serializer(), Serdes.String().serializer());
+            final TestOutputTopic<String, String> outputTopic =
+                driver.createOutputTopic(output, Serdes.String().deserializer(), Serdes.String().deserializer());
+            final TestOutputTopic<String, String> streamToTableOutputTopic =
+                driver.createOutputTopic(streamToTableOutput, Serdes.String().deserializer(), Serdes.String().deserializer());
+
+            inputTopic.pipeInput("A", "01", 5L);
+            inputTopic.pipeInput("B", "02", 100L);
+            inputTopic.pipeInput("C", "03", 0L);
+            inputTopic.pipeInput("D", "04", 0L);
+            inputTopic.pipeInput("A", "05", 10L);
+            inputTopic.pipeInput("A", "06", 8L);
+
+            streamToTableInputTopic1.pipeInput("A", "01", 5L);
+            streamToTableInputTopic1.pipeInput("B", "02", 100L);
+            streamToTableInputTopic1.pipeInput("C", "03", 0L);
+            streamToTableInputTopic1.pipeInput("D", "04", 0L);
+            streamToTableInputTopic1.pipeInput("A", "05", 10L);
+            streamToTableInputTopic1.pipeInput("A", "06", 8L);
+            assertEquals(outputTopic.readRecordsToList(), streamToTableOutputTopic.readRecordsToList());
+        }
+    }
+
+    @Test
+    public void shouldKStreamToKTableJoinAndMaterializedStore() {
+        final Consumed<String, String> consumed = Consumed.with(Serdes.String(), Serdes.String());
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final String input1 = "input1";
+        final String input2 = "input2";
+        final String output = "output";
+        final String storeName = "store";
+
+        final KTable table1 = builder.stream(input1, consumed)
+            .toTable(Materialized.as(Stores.inMemoryKeyValueStore(storeName)));
```

Comment:
This test should pass without the need to pass in `Materialized` object to force a materialization explicitly (ie, when we call join later (btw: we should also add tests for table-table and foreign-key-join; both with using the "stream-table" as left or right input).

To test a forces materialization, we should have a separate test:
```builder.stream(input1, consumed).toTable(Materialized.as(Stores.inMemoryKeyValueStore(storeName)));```

That verifies the result by querying the store only (no output topic would be required for this test).